docs/Encryption.md

```diff
@@ -0,0 +1,76 @@
+S3Proxy
+
+# Encryption 
+
+## Motivation 
+The motivation behind this implementation is to provide a fully transparent and secure encryption to the s3 client while having the ability to write into different clouds.
+
+## Cipher mode
+The chosen cipher is ```AES/CFB/NoPadding``` because it provides the ability to read from an offset like in the middle of a ```Blob```.
```

Comment:
Why AES/CFB on 128 bits instead of the nowadays very common AES 256 GCM ?

Reply:
@fungiboletus as I wrote AES CFB allows to us to read from an offset without having to get in a worst case multiple gigabytes to only read some needed bytes.
As of now I don't saw an implementation how to read from a specific offset of a file with AES GCM but feel free to fork this PR and propose the change. Also we can handle multiple versions of the encryption in the block header so we can also change it somewhere in the future

Reply:
GCM uses a counter of so it's relatively easy to start from an offset, you lose the authentication though. For that you need to process the whole file. For example ZFS uses AES-256-GCM by default for encryption, so it is working well in practice.

I was just curious about your choices, I'm unfortunately not very confident in crypto in Java and I see in your code that this is quite low level. It's not about changing a cipher mode in a constant somewhere 😅

Anyway your work is very appreciated 🙂